### PR TITLE
Packer base HM 5

### PR DESCRIPTION
**Pull request by palekseym**
# Выполнено ДЗ № 5

 - [X] Основное ДЗ
 - [X] Задание со *

## В процессе сделано:
 - Установлен packer
 - Создан шаблон reddit-base ubuntu16.json, чувствительные данные вынесены в файл переменных variables.json
 - Создан шаблон reddit-full, на базе шаблона из первого пункта,  для практики подхода Immutable infrastructure. шаблон immutable.json, файл переменных  variables-immutable.json
 - Создан скрипт create-redditvm.sh для создания виртуальной машины из шаблона reddit-full
 - Настроен автоспуск приложения в образе reddit-full. Unit для systemd файл лежит в packer/files/reddit.service

### Подготовить образы
 - Запустить в директории packer команду
`packer build --var-file variables.json ubuntu16.json`
 - Запустить в директории packer команду
`packer build --var-file variables-immutable.json immutable.json`
### Создать виртуальную машину
 - Запустить в директории config-scripts команду
`./create-redditvm.sh`
Результат
```
tay@ubuntu:~/repo/palekseym_infra/config-scripts$ ./create-redditvm.sh
Created [https://www.googleapis.com/compute/v1/projects/sapient-cycling-225707/zones/europe-west4-c/instances/reddit-app].
NAME        ZONE            MACHINE_TYPE  PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP    STATUS
reddit-app  europe-west4-c  g1-small                   10.164.0.11  35.204.67.210  RUNNING

```
## Как проверить работоспособность:
 - Перейти по ссылке http://35.204.67.210:9292
 - Также можно открыть используя `curl -X GET http://35.204.67.210:9292`

```
<!DOCTYPE html>
<html lang='en'>
<head>
<meta charset='utf-8'>
<meta content='IE=Edge,chrome=1' http-equiv='X-UA-Compatible'>
<meta content='width=device-width, initial-scale=1.0' name='viewport'>
<title>Monolith Reddit :: All posts</title>
<link crossorigin='anonymous' href='https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css' integrity='sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME$
fgjWPGmkzs7' rel='stylesheet' type='text/css'>
<link crossorigin='anonymous' href='https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap-theme.min.css' integrity='sha384-fLW2N01lMqjakBkx3l/M9EahuwpSfeNvV63J5ezn3uZzap$
0u7EYsXMjQV+0En5r' rel='stylesheet' type='text/css'>
<script crossorigin='anonymous' href='https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js' integrity='sha384-0mSbJDEHialfmuBBQP6A4Qrprq5OVfW37PRR3j5ELqxss1yVqOtn$
pnHVP9aJ7xS'></script>
<script src='https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js'></script>
</head>
<body>
<div class='navbar navbar-default navbar-static-top'>
<div class='container'>
<div class='navbar-header'>
<button class='navbar-toggle' data-target='.navbar-responsive-collapse' data-toggle='collapse' type='button'>
<span class='icon-bar'></span>
<span class='icon-bar'></span>
<span class='icon-bar'></span>
</button>
<a class='navbar-brand' href='/'>Monolith Reddit</a>
</div>
<div class='navbar-collapse collapse'>
<ul class='nav navbar-nav navbar-right'>
<li>
<a href='/signup'>Sign up</a>
</li>
<li>
<a href='/login'>Login</a>
</li>
</ul>
</div>
</div>
</div>
<div class='container'>
<div class='row'>
<div class='col-lg-9'>

</div>
<div class='col-lg-3'>
<div class='well sidebar-nav'>
<h3>Menu</h3>
<ul class='nav nav-list'>
<li>
<a href='/'>All posts</a>
</li>
<li>
<a href='/new'>New post</a>
</li>
</ul>
</div>
</div>
</div>
</div>
</body>
</html>
```

## PR checklist
 - [X] Выставил label с темой домашнего задания

### CODE REVIEW

**Review by enkov — APPROVED** [review]
Добрый день!
Спасибо за работу. Задание принято.
